### PR TITLE
Update Kubespray to HEAD

### PR DESCRIPTION
**Pull request by NicolasT**
Fixes: #364
Fixes: https://github.com/scality/metalk8s/issues/364

### CODE REVIEW

**Comment by NicolasT** [comment]
Mainly looking for CI feedback. Need to update ChangeLog, at least.

Running an upgrade on my test cluster as we speak.

**Comment by NicolasT** [comment]
@bert-e wait

**Comment by NicolasT** [comment]
This is a massive change (which means we really should try to stick closer to upstream...). We do have some testing in CI now, though I really think this needs 'actual review'.

I've gone over the changes already, will do so again. Been running into some troubles on my install, though that one is kinda broken due to #302 testing, so not sure it's an actual issue.

Would be useful if others could try to run `deploy.yml` on a pre-existing cluster...

File: vendor/kubespray/.gitlab-ci.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ before_script:
     # Check out latest tag if testing upgrade
     # Uncomment when gitlab kubespray repo has tags
     #- test "${UPGRADE_TEST}" != "false" && git fetch --all && git checkout $(git describe --tags $(git rev-list --tags --max-count=1))
-    - test "${UPGRADE_TEST}" != "false" && git checkout f7d52564aad2ff8e337634951beb4a881c0e8aa6
+    - test "${UPGRADE_TEST}" != "false" && git checkout 8b3ce6e418ccf48171eb5b3888ee1af84f8d71ba
```
Comment (Zempashi) [inline vendor/kubespray/.gitlab-ci.yml:96]: The upgrade seems to work accross tags. Shouldn't we use tag version instead of master ?

Comment (NicolasT) [reply vendor/kubespray/.gitlab-ci.yml:96]: We could, but haven't before. Kubespray releases tend to be... somewhat ad-hoc.

**Review by Zempashi — APPROVED** [review]

**Comment by NicolasT** [comment]
Thx for the review @Zempashi

**Pull request merged**